### DISCUSSION TITLE
Flash HomeSpan onto existing smart home devices

### DISCUSSION THREAD

**Discussion by cmyplay** [HomeSpan/HomeSpan#851] [Q&A]
What would it take to flash HomeSpan onto an existing ESP32 smart home device?

Has anyone done it?
Is it even possible?

**Comment by G-D-D-971** [comment]
I think it's possible if the "smart home device" use an esp32 chip…

**Comment by HomeSpan** [comment]
I don't see how this would work in practice.   Any commercial system using an ESP32 would likely have embedded software tuned to the electronics it was using.   You would need to know exactly how the chip was used, how connections were made, etc., and the device would need to be sufficiently open for you to upload your own software.  I'm not aware of any commercial devices that allow this -- this might be possible on a semi-commercial open source device that provided schematics, etc.

**Reply by G-D-D-971** [reply]
Yes it's possible for example you can flash ESPhome on Tuya device, but after it's ESPhome, I don't know for HomeSpan…
